libasynql/src/poggit/libasynql/mysqli/MysqlSslCredentials.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * libasynql
+ *
+ * Copyright (C) 2018 SOFe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace poggit\libasynql\mysqli;
+
+use JsonSerializable;
+use mysqli;
+
+class MysqlSslCredentials implements JsonSerializable{
+	/** @var string $key */
+	private $key;
+	/** @var string $certificate */
+	private $certificate;
+	/** @var string $caCertificate */
+	private $caCertificate;
+	/** @var string $caPath */
+	private $caPath;
+	/** @var string $cipherAlgorithms */
+	private $cipherAlgorithms;
+
+
+	/**
+	 * Creates a new {@link MysqlSslCredentials} instance from an array (e.g. from Config), with empty default values.
+	 * @param array       $array
+	 * @return MysqlSslCredentials
+	 */
+	public static function fromArray(array $array) : MysqlSslCredentials{
+		return new MysqlSslCredentials(
+			$array["key"] ?? "",
+			$array["certificate"] ?? "",
+			$array["ca-certificate"] ?? "",
+			$array["ca-path"] ?? "",
```

Comment:
These should be nullable. Not familiar with mysqli, but why do we need both caCertificate and caPath? What is the different between them? The typical CA bundle only involves key + cert + CA afaik.

Reply:
This is just a loose binding to mysqli's `ssl_set` function. I can remove those parameters if you'd prefer? Regardless, I will make them nullable.